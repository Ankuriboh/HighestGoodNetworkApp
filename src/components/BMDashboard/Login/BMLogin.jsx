const BMLogin = () => {
  return <div>Login here</div>
}

export default BMLogin