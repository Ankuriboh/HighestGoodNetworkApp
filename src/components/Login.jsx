import React from "react";
import Joi from "joi";
import Form from "./common/form";
import { login, getCurrentUser } from "../services/loginService";
import logService from '../services/logService';
import {Redirect} from 'react-router-dom'

class Login extends Form {
  state = {
    data: { email: "", password: "" },
    errors: {}
  };

  schema = {
    email: Joi.string()
      .email()
      .required()
      .label("Email"),
    password: Joi.string()
      .required()
      .label("Password")
  };

  doSubmit = async () => {
    const email = this.state.data.email;
    const password = this.state.data.password;
    try {
    await login({ email, password });  
    const {state}   = this.props.location;
     window.location = state? state.from.pathname : "/"
    } catch (ex) {
      logService.logError(ex);
      logService.logInfo(ex.response)
      //logService.logInfo(ex.response.status)
      if(ex.response && ex.response.status === 403)
      {       
        
        const errors = this.state.errors;
        errors["email"] = ex.response.data.message;//"Invalid email and/ or password.";
        this.setState({errors})
      }
    }
  };

  render() {
if (getCurrentUser()) 
return <Redirect to ="/"/>

    return (

      <div className = "container mt-5">
      <h2>Please Sign in</h2>
     
      <form className="col-md-6 xs-12" onSubmit={e => this.handleSubmit(e)}>
        {this.renderInput("email", "Email:")}
        {this.renderInput("password", "Password:", "password")}
        {this.renderButton("Submit")}
      </form>
     
      </div>
    );
  }
}

export default Login;