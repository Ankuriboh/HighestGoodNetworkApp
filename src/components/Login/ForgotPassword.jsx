import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { Button, Input } from 'reactstrap';
import { forgotPassword } from '../../services/authorizationService';


const ForgotPassword = React.memo(() => {
  debugger;
  const [email, onEmailChange] = useState('');
  const [firsName, setFirstName] = useState(undefined);
  const [lastName, setLastName] = useState(undefined);


  const onForgotPassword = () => {
    let forgotPasswordData = { emailId: email, firstname: firsName, lastname: lastName };
    forgotPassword(forgotPasswordData).then(response => {
      alert("NewPassword sent to your mail id");
    }).catch(Err => { alert('failed') })
  }

  return (

    <div className="container mt-5">
      <form className="col-md-6 xs-12">

        <label>Email</label>
        <Input type='text' placeholder='enter your email ID'
          value={email}
          onChange={(e) => { onEmailChange(e.target.value) }} />

        <label>First Name</label>
        <Input type='text' placeholder='enter your first name'
          value={firsName}
          onChange={(e) => { setFirstName(e.target.value) }} />

        <label>Last Name</label>
        <Input type='text' placeholder='enter your last name'
          value={lastName}
          onChange={(e) => { setLastName(e.target.value) }} />

        <div style={{ marginTop: '40px' }}>
          <Button color='primary' onClick={onForgotPassword}>Submit</Button>
          <Link to='login'>
            <Button style={{ marginLeft: '350px' }}>Cancel</Button>
          </Link>
        </div>

      </form>
    </div>
  )
})
export default ForgotPassword