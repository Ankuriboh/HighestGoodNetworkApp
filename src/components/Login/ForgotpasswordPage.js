import React, { Component } from 'react';
import axios from 'axios';
import { Link, Redirect } from "react-router-dom";

class ForgotPassword extends Component {
    constructor() {
        super();
        this.state = {
            firstName: '',
            lastName: '',
            email: '',
            validate: {
                emailState: ""
            },
            redirect:false
        }
        this.handleChangeandValidate = this.handleChangeandValidate.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.submitForm=this.submitForm.bind(this);
    }
    handleChangeandValidate(event) {
        this.handleChange(event);
        this.validateEmail(event);
    }
    validateEmail(e) {
        const emailRex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
        const { validate } = this.state;
        if (emailRex.test(e.target.value)) {
            validate.emailState = "has-success";
        } else {
            validate.emailState = "has-danger";
        }
        this.setState({ validate });
    }

    handleChange = async event => {
        const { target } = event;
        const value = target.value;
        const { name } = target;
        await this.setState({
            [name]: value
        });
    };
    submitForm(e) {
        e.preventDefault();

        axios({
            method: 'post',
            url: 'http://localhost:4500/api/forgotpassword',
            data: {
              email: this.state.email,
              firstName: this.state.firstName,
              lastName:this.state.lastName
            },
            headers: {
              'Content-Type': 'application/json'
            }
        })
        .then(
            () => {
              alert('you will get an email with new password shortly');
              this.setState({
                redirect:true
                
              })
            }, error => {
              alert("something went wrong");
            }
          )
    }

    renderRedirect = () => {
        if (this.state.redirect) {
          return <Redirect to='/login' />
        }
    }
    render() {
        const { email, firstName, lastName } = this.state;
        return ( null
            // <div>
            //     {this.renderRedirect()}
            // <Container className="forgotpassword" >
            //     <Form className="form formforgotPwd" onSubmit={e => this.submitForm(e)}>
            //         <h3 className=" portal py-4 text-center">
            //             One Community Global Forgot Password Form
            //     </h3>
            //         <FormGroup>
            //             <Input
            //                 required
            //                 type="text"
            //                 name="firstName"
            //                 id='firstName'
            //                 placeholder='firstName'
            //                 value={firstName}
            //                 onChange={this.handleChange}
            //             />
            //             <FormFeedback>firstName must atleast be two char length</FormFeedback>
            //         </FormGroup>
            //         <FormGroup>
            //             <Input
            //                 required
            //                 type="text"
            //                 name="lastName"
            //                 id='lastName'
            //                 placeholder='lastName'
            //                 value={lastName}
            //                 onChange={this.handleChange}
            //             />
            //             <FormFeedback>Name must atleast be two char length</FormFeedback>
            //         </FormGroup>
            //         <FormGroup>
            //             <Input
            //                 type="email"
            //                 name="email"
            //                 id="email"
            //                 placeholder="Email"
            //                 value={email}
            //                 valid={this.state.validate.emailState === "has-success"}
            //                 invalid={this.state.validate.emailState === "has-danger"}
            //                 onChange={this.handleChangeandValidate}
            //             />
            //             <FormFeedback>Invalid email address</FormFeedback>
            //         </FormGroup>
            //         <Row>
            //             <Col>
            //                 <Link to="/login" className="text-muted">
            //                     Cancel
            //                 </Link>
            //             </Col>
            //             <Col>
            //                 <Button color="success" block >
            //                     Submit
            //                 </Button>
            //             </Col>
            //         </Row>
            //     </Form>
            // </Container>
            // </div>
        );
    }
}

export default ForgotPassword;