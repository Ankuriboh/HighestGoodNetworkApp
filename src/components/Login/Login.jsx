import React from "react";
import Joi from "joi";
import Form from "../common/Form";
import { connect } from "react-redux";
import { getCurrentUser } from "../../services/loginService";
import { loginUser } from "../../actions/index"
import { Redirect } from "react-router-dom";

class Login extends Form {
  state = {
    data: { email: "", password: "" },
    errors: {}
  };

  schema = {
    email: Joi.string()
      .email()
      .required()
      .label("Email"),
    password: Joi.string()
      .required()
      .label("Password")
  };

  componentDidMount() {
    document.title = "Login";
    if (this.state.user && this.state.user.role){
      window.location = '/';
    }
  }

  componentDidUpdate(prevProps) {
    if (prevProps.state.user !== this.props.state.user && this.props.state.user !== null){
      if (this.props.state.user.errors){
        this.setState({ errors: this.props.state.user.errors });
      } 
      else if (this.props.state.user.new) {
        window.location = `/forcePasswordUpdate/${this.props.state.user.userId}`;
      } 
      else{
        const { state } = this.props.location;
        window.location = state ? state.from.pathname : "/dashboard";
      }
    }
  }

  doSubmit = async () => {
    const email = this.state.data.email;
    const password = this.state.data.password;
    // try {
    this.props.loginUser({ email, password });
      // if (result && result.userType === "newUser") {
      //   window.location = `/forcePasswordUpdate/${result.userId}`;
      //   return;
      // }
    //   const { state } = this.props.location;
    //   window.location = state ? state.from.pathname : "/dashboard";
    //   return;
    // } catch (ex) {
    //   if (ex.response && ex.response.status === 403) {
    //     const errors = this.state.errors;

    //     errors["email"] = ex.response.data.message;
    //     this.setState({ errors });
    //   }
    // }
  };

  render() {
    // if (getCurrentUser()) return <Redirect to="/" />;

    return (
      <div className="container mt-5">
        <h2>Please Sign in</h2>

        <form className="col-md-6 xs-12" onSubmit={e => this.handleSubmit(e)}>
          {this.renderInput({ name: "email", label: "Email:" })}
          {this.renderInput({
            name: "password",
            label: "Password:",
            type: "password"
          })}
          {this.renderButton("Submit")}
        </form>
      </div>
    );
  }
}


const mapStateToProps = state => {
  return { state };
};

export default connect(mapStateToProps, { 
  loginUser
})(Login);
