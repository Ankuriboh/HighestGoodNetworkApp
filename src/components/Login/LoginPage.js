import React from "react";
import Joi from "joi";
import Form from "../../common/form";
import { login, getCurrentUser } from "../../services/loginService";
import {Redirect} from 'react-router-dom'

class LoginForm extends Form {
  state = {
    data: { email: "", password: "" },
    errors: {}
  };

  schema = {
    email: Joi.string()
      .email()
      .required()
      .label("Email"),
    password: Joi.string()
      .required()
      .label("Password")
  };

  doSubmit = async () => {
    const email = this.state.data.email;
    const password = this.state.data.password;
    try {
    await login({ email, password });  
    const {state}   = this.props.location;
     window.location = state? state.from.pathname : "/"
    } catch (ex) {
      if(ex.response && ex.response.status === 403)
      {       
        const errors = this.state.errors;
        errors["email"] = ex.response.data.message
        this.setState({errors})
      }
    }
  };

  render() {
if (getCurrentUser()) 
return <Redirect to ="/"/>

    return (

      <div className = "container mt-5">
      <h2>Please Sign in</h2>
     
      <form className="col-md-6 xs-12" onSubmit={e => this.handleSubmit(e)}>
        {this.renderInput("email", "Email:")}
        {this.renderInput("password", "Password:")}
        {this.renderButton("Submit")}
      </form>
     
      </div>
    );
  }
}

export default LoginForm;