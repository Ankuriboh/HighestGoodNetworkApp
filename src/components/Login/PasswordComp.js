import React,{Component} from 'react';
import { FormGroup, Input, FormFeedback} from 'reactstrap';

class PasswordComp extends Component {
  constructor(){
    super();
    this.state={
      password:""
    }
    this.handleChange=this.handleChange.bind(this);
  }

  handleChange = async event => {
    const { target } = event;
    const value = target.value;
    const { name } = target;
    await this.setState({
      [name]: value
    });
  };
  render() {
    const { password } = this.state;
    const {id} = this.props;
    return (
      <div>
        <FormGroup>
          <Input
            required
            type="password"
            name="password"
            id={id}
            placeholder={id}
            value={password}
            onChange={this.handleChange}
          />
        </FormGroup>
        <FormFeedback>Password should be minimum of 8 characters with atleast one capital,one small letter, one special charac,one number </FormFeedback>
      </div>
    );
  }
}
export default PasswordComp;
