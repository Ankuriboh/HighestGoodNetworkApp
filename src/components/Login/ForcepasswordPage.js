import React,{Component} from 'react';
import PasswordComp from './PasswordComp';
import {Container} from 'reactstrap';
import  {Button,FormFeedback} from "reactstrap";
import { Redirect } from "react-router-dom";
import axios from 'axios';

const container = {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    flexFlow: 'column'
  }

class ForcePasswordPage extends Component {
    constructor(){
        super();
        this.state={
            redirect:false
        }
        this.submit =this.submit.bind(this);
        this.renderRedirect=this.renderRedirect.bind(this);
    }

    submit(e){
        e.preventDefault();
        axios({
            method: 'patch',
            url: 'http://localhost:4500/api/forcepassword',
            data: {
              userId: this.state.email,
              newpassword: this.state.password
            },
            headers: {
              'Content-Type': 'application/json'
            }
        }).then(
            () => {
                alert('use the new password set to login to your account');
                this.setState({
                    redirect: true
                })
            }, error => {
              alert("something went wrong");
            }
          )
    }
    renderRedirect(){
        if(this.state.redirect){
          return <Redirect to='/login' />
        }
    }
    render() {
        return (
            <div>
                {this.renderRedirect()}
            <Container className="forcePassword" style={container}>
            <h3 className="py-4 text-center">
            Change Password
            </h3>
            <PasswordComp id='password' />
            <PasswordComp id='confirmPassword'/>
            <FormFeedback>Both the paswords must</FormFeedback>
            <Button color="success" block className="py-2" onClick={this.submit}>
                 Submit
            </Button>
          
            </Container>
            </div>
        );
    }
}

export default ForcePasswordPage;