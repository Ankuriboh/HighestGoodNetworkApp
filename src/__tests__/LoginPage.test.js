import React from 'react';
import { shallow } from 'enzyme';
import Login from '../components/Login';

describe("Basic Structure for Login", () => {
    
    let mountedLogin;
    beforeEach(() => {
        mountedLogin = shallow(<Login />);
    })

    it('Login is rendered with two input fields', () => {
        
        const inputs = mountedLogin.find('Input')
        expect(inputs.length).toBe(2)
      });

    it("Login is rendered with one button", () => 
    {
        
        const button = mountedLogin.find("button.btn[disabled]")
        expect(button.length).toBe(1)
        
        
    })
    it("Login is rendered with one h2", () => 
    {
        
        const h2 = mountedLogin.find("h2")
        expect(h2.length).toEqual(1)
        expect(h2.first().text()).toContain("Please Sign in")
        
    })
      
})

describe("When user tries to login", () => {

    let mountedLoginPage;
    let state;

    beforeEach(() => {
        mountedLoginPage = shallow(<Login />);
        })

    
    it("should call handleChnage when input is changed", () => {
        const spy = jest.spyOn(mountedLoginPage.instance(), 'handleChange');
        mountedLoginPage.find("#email").simulate('change', {currentTarget : {name: "email", value : "anc"}})
        expect(spy).toHaveBeenCalled();            
        })

    it("should correctly update the email value in the state", () => {
        let expected =   "sh@gmail.com"
         let Input =  {name: "email", "value": expected };
         let mockEvent = {currentTarget: Input}
         mountedLoginPage.instance().handleChange(mockEvent);
      
       expect(mountedLoginPage.instance().state.data.email).toEqual(expected)
        
    })

    it("should correctly update the error if the email is not in correct format", () => { 
        let expected =   "sh";            
        let Input =  {name: "email", "value": expected};
        let mockEvent = {currentTarget: Input}
        mountedLoginPage.instance().handleChange(mockEvent);     
      expect(mountedLoginPage.instance().state.errors["email"]).toEqual('"Email" must be a valid email');
   })


   it("should correctly update the password value in the state", () => {
    let expected =   "trapp"
     let Input =  {name: "password", "value": expected };
     let mockEvent = {currentTarget: Input}
     mountedLoginPage.instance().handleChange(mockEvent);
  
   expect(mountedLoginPage.instance().state.data.password).toEqual(expected)
    
})

it("should correctly update the errors object if the password is empty", () => { 
    let expected =   "";            
    let Input =  {name: "password", "value": expected};
    let mockEvent = {currentTarget: Input}
    mountedLoginPage.instance().handleChange(mockEvent); 
    expect(mountedLoginPage.instance().state.errors["password"]).toEqual('"Password" is not allowed to be empty');     
 
})

it("should have disabled submit button if form in invalid", () => { 
    const button = mountedLoginPage.find("button")       
    expect(button.props()).toHaveProperty("disabled");  
})

it("form can be submitted", () => {
    const callback = jest.fn();
    let mountedLoginPagewithCallBack = shallow(<form onSubmit = {callback}/>)     
     mountedLoginPagewithCallBack.find("form").simulate("submit");
    expect(callback).toHaveBeenCalled()

})

it("onSubmit login method is called", async()=> {

    const loginService = require("../services/loginService")
    let syplogin = jest.spyOn(loginService, 'login');    
    await mountedLoginPage.instance().doSubmit();
    expect(syplogin).toHaveBeenCalled();
})

})