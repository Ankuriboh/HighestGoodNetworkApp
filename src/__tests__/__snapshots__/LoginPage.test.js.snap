// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page structure should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Please Sign in
        </h2>,
        <form
          className="col-md-6 xs-12"
          onSubmit={[Function]}
        >
          <Input
            id="email"
            label="Email:"
            name="email"
            onChange={[Function]}
            type="text"
            value=""
          />
          <Input
            id="password"
            label="Password:"
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          />
          <button
            className="btn btn-primary"
            disabled={
              Object {
                "email": "\\"Email\\" is not allowed to be empty. \\"Email\\" must be a valid email",
                "password": "\\"Password\\" is not allowed to be empty",
              }
            }
          >
            Submit
          </button>
        </form>,
      ],
      "className": "container mt-5",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Please Sign in",
        },
        "ref": null,
        "rendered": "Please Sign in",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Input
              id="email"
              label="Email:"
              name="email"
              onChange={[Function]}
              type="text"
              value=""
            />,
            <Input
              id="password"
              label="Password:"
              name="password"
              onChange={[Function]}
              type="password"
              value=""
            />,
            <button
              className="btn btn-primary"
              disabled={
                Object {
                  "email": "\\"Email\\" is not allowed to be empty. \\"Email\\" must be a valid email",
                  "password": "\\"Password\\" is not allowed to be empty",
                }
              }
            >
              Submit
            </button>,
          ],
          "className": "col-md-6 xs-12",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "error": undefined,
              "id": "email",
              "label": "Email:",
              "name": "email",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "error": undefined,
              "id": "password",
              "label": "Password:",
              "name": "password",
              "onChange": [Function],
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "className": "btn btn-primary",
              "disabled": Object {
                "email": "\\"Email\\" is not allowed to be empty. \\"Email\\" must be a valid email",
                "password": "\\"Password\\" is not allowed to be empty",
              },
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Please Sign in
          </h2>,
          <form
            className="col-md-6 xs-12"
            onSubmit={[Function]}
          >
            <Input
              id="email"
              label="Email:"
              name="email"
              onChange={[Function]}
              type="text"
              value=""
            />
            <Input
              id="password"
              label="Password:"
              name="password"
              onChange={[Function]}
              type="password"
              value=""
            />
            <button
              className="btn btn-primary"
              disabled={
                Object {
                  "email": "\\"Email\\" is not allowed to be empty. \\"Email\\" must be a valid email",
                  "password": "\\"Password\\" is not allowed to be empty",
                }
              }
            >
              Submit
            </button>
          </form>,
        ],
        "className": "container mt-5",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Please Sign in",
          },
          "ref": null,
          "rendered": "Please Sign in",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Input
                id="email"
                label="Email:"
                name="email"
                onChange={[Function]}
                type="text"
                value=""
              />,
              <Input
                id="password"
                label="Password:"
                name="password"
                onChange={[Function]}
                type="password"
                value=""
              />,
              <button
                className="btn btn-primary"
                disabled={
                  Object {
                    "email": "\\"Email\\" is not allowed to be empty. \\"Email\\" must be a valid email",
                    "password": "\\"Password\\" is not allowed to be empty",
                  }
                }
              >
                Submit
              </button>,
            ],
            "className": "col-md-6 xs-12",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "error": undefined,
                "id": "email",
                "label": "Email:",
                "name": "email",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "error": undefined,
                "id": "password",
                "label": "Password:",
                "name": "password",
                "onChange": [Function],
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "className": "btn btn-primary",
                "disabled": Object {
                  "email": "\\"Email\\" is not allowed to be empty. \\"Email\\" must be a valid email",
                  "password": "\\"Password\\" is not allowed to be empty",
                },
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
